server/controllers/document.js

```diff
@@ -0,0 +1,99 @@
+import model from '../models/index';
+
+const Document = model.Document;
+
+export default {
+  createDoc(req, res) {
+    req.body.UserId = req.decoded.UserId;
+    return Document
+      .create(req.body)
+      .then(document => res.status(201).send(document))
+      .catch((error) => {
+        res.status(500).send(error);
+      });
+  },
+
+  listDocs(req, res) {
+    return Document
+    .all()
+    .then(document => res.status(200).send(document));
+  },
+
+  getDoc(req, res) {
+    return Document
+    .findById(req.params.id)
+    .then((document) => {
+      if (!document) {
+        return res.status(404).send({
+          message: 'Document Not Found'
+        });
+      }
+      if (document.RoleId === req.decoded.RoleId || document.UserId === req.decoded.UserId) {
+        return res.status(200).send(document);
+      }
+      res.status(403).send({
+        message: 'You cannot view this document'
+      });
+    });
+  },
+
+  getRoleDoc(req, res) {
+    return Document
+    .findAll({ where: { RoleId: req.query.RoleId } })
+    .then(document => res.status(200).send(document));
+  },
+
+  getUserDoc(req, res) {
+    return Document
+    .findAll({ where: { UserId: req.query.UserId } })
+    .then(document => res.status(200).send(document));
+  },
+
+  getDateDoc(req, res) {
+    return Document
+    .findAll({ where: { createdAt: { $contain: req.query.date } } })
+    .then(document => res.status(200).send(document));
+  },
+
+  updateDoc(req, res) {
+    return Document
+    .findById(req.params.id, {})
+    .then(document => {
+      if (!document) {
+        return res.status(404).send({
+          message: 'Document Not Found',
+        });
+      }
+      if (document.UserId !== req.decoded.UserId) {
+        return res.status(403).send({
+          message: 'You cannot update this document'
+        });
+      }
+      return document
+        .update(req.body)
```

Comment:
You can use `where` in the `.update options` parameter. It will go a long way in eliminating the need to find the document first before update. Check [here](http://docs.sequelizejs.com/en/v3/api/model/#updatevalues-options-promisearrayaffectedcount-affectedrows) for the documentation